src/eos-updater-extensions.c

```diff
@@ -0,0 +1,129 @@
+}
+
+static void
+eos_extensions_finalize_impl (EosExtensions *extensions)
+{
+  g_free (extensions->ref_name);
+}
+
+EOS_DEFINE_REFCOUNTED (EOS_EXTENSIONS,
+                       EosExtensions,
+                       eos_extensions,
+                       eos_extensions_dispose_impl,
+                       eos_extensions_finalize_impl)
+
+EosExtensions *
+eos_extensions_new_empty (void)
+{
+  return g_object_new (EOS_TYPE_EXTENSIONS, NULL);
+}
+
+EosExtensions *
+eos_extensions_new (OstreeRepo *repo,
+                    GCancellable *cancellable,
+                    GError **error)
+{
+  g_autoptr(EosBranchFile) branch_file = NULL;
+  g_autoptr(EosExtensions) extensions = NULL;
+
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), NULL);
+
+  branch_file = eos_branch_file_new_from_repo (repo,
+                                               cancellable,
+                                               error);
+  if (branch_file == NULL)
+    return NULL;
+
+  extensions = eos_extensions_new_empty ();
+  extensions->branch_file = g_steal_pointer (&branch_file);
+  return g_steal_pointer (&extensions);
+}
+
+gboolean
+eos_extensions_save (EosExtensions *extensions,
+                     OstreeRepo *repo,
+                     GCancellable *cancellable,
+                     GError **error)
+{
+  g_autoptr(GFile) ext_path = NULL;
+  g_autoptr(GFile) branch_file_path = NULL;
+  g_autoptr(GFile) branch_file_sig_path = NULL;
+
+  g_return_val_if_fail (EOS_IS_EXTENSIONS (extensions), FALSE);
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
+
+  if (!eos_updater_create_extensions_dir (repo, &ext_path, error))
+    return FALSE;
+  if (!eos_updater_save_or_delete (extensions->summary,
+                                   ext_path,
+                                   "summary",
+                                   cancellable,
+                                   error))
+    return FALSE;
+
+  if (!eos_updater_save_or_delete (extensions->summary_sig,
+                                   ext_path,
+                                   "summary.sig",
+                                   cancellable,
+                                   error))
+    return FALSE;
+
+  if (extensions->ref_name != NULL)
+    {
+      g_autofree gchar *ref_sig_filename = NULL;
+
+      if (!eos_updater_save_or_delete (extensions->summary_sig,
+                                       ext_path,
+                                       extensions->ref_name,
+                                       cancellable,
+                                       error))
+        return FALSE;
+
+      ref_sig_filename = g_strconcat (extensions->ref_name, ".sig", NULL);
+      if (!eos_updater_save_or_delete (extensions->summary_sig,
+                                       ext_path,
+                                       ref_sig_filename,
+                                       cancellable,
+                                       error))
+        return FALSE;
```

Comment:
Yes, it was, but I'm not sure why github is still showing it as non-outdated diff…

Actually I just made another change here (forgot that we should put the refs file in the `refs.d` directory).